File: src/client/xnative/lion.xdroid/Service/PushReceiverService.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Lion.XDroid;
using Firebase.Messaging;
using Lion.XDroid.Libs;

namespace Lion.XDroid.Service
{
    [Service]
    [IntentFilter(new[] { "com.google.firebase.MESSAGING_EVENT" })]
    public class PushReceiverService : FirebaseMessagingService
    {
        public override void OnMessageReceived(RemoteMessage remoteMessage)
        {
            var data = remoteMessage.Data;

            var _builder = (Notification.Builder)null;
            if (Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
                _builder = new Notification.Builder(this, "lion_default_channel");
            else
#pragma warning disable CS0618 // Type or member is obsolete
                _builder = new Notification.Builder(this);
#pragma warning restore CS0618 // Type or member is obsolete


            var notifyBuilder = _builder
                                    .SetContentTitle(data["title"])
                                    .SetContentText(data["message"])
                                    .SetSmallIcon(Resource.Drawable.alarm)
                                    .SetWhen(Java.Lang.JavaSystem.CurrentTimeMillis());

            var _notification_manager = this.ApplicationContext.GetSystemService(Context.NotificationService).JavaCast<NotificationManager>();
            _notification_manager.Notify(0, notifyBuilder.Build());

            AppCommon.SNG.ResetBadge(this);
        }
    }
}

File: src/client/xcode/Lion.Droid/app/src/main/java/com/odinsoftware/lottolion/service/PushReceiverService.java
package com.odinsoftware.lottolion.service;

import android.app.Notification;
import android.app.NotificationManager;
import android.graphics.Color;
import android.os.Build;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.odinsoftware.lottolion.R;

import java.util.Map;

/**
 * Created by limyg on 2017-04-27.
 */
public class PushReceiverService extends FirebaseMessagingService {
    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        Map<String, String> data = remoteMessage.getData();
        Notification.Builder notifyBuilder = new Notification.Builder(this)
                .setContentTitle(data.get("title"))
                .setContentText(data.get("message"))
                .setSmallIcon(R.drawable.alarm)
                .setWhen(System.currentTimeMillis());

        NotificationManager notificationManager =
                (NotificationManager) getSystemService(this.NOTIFICATION_SERVICE);
        notificationManager.notify(0, notifyBuilder.build());

        LLCommon.Get().ResetBadge(this);
    }
}
